### ISSUE TITLE
False positive: Not all code paths of <function/method> end in a return statement, return type <type> expected

### ISSUE THREAD

**Issue by eclipxe13** [vimeo/psalm#2183]
This snippet produces the following error, using psalm 3.5.3 & psalm.dev

```
ERROR: InvalidReturnType - 3:17 - Not all code paths of foo end in a return statement, return type int expected
```

```php
<?php

function foo(): int
{
  do {
    $value = mt_rand(0, 10);
    if ($value > 5) continue;
    break;
  } while (true);
  return $value;
}

function bar(): int
{
  $psalmContinue = true;
  do {
    $value = mt_rand(0, 10);
    if ($value > 5) continue;
    $psalmContinue = false;
  } while ($psalmContinue);
  return $value;
}

function baz(): void
{
  do {
    $value = mt_rand(0, 10);
    if ($value > 5) continue;
    break;
  } while (true);
}
```

In the previous example `foo`, `bar` & `baz` do the same thing, but using `break` in `foo` produces the false positive error. It is not present when there is no return type.

**Comment by eclipxe13** [comment]
Wow. Thanks
